### PR TITLE
Live Submission GUI

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #84 

image

### CODE REVIEW

**Review by nyngwang — APPROVED** [review]
LGTM

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Table header's color should be `#676877` 
thanks :)

**Review by Johnny850807 — APPROVED** [review]
Review done

File: src/app/exam/submission/ExamSubmission.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import FakeLink from "../../commons/FakeLink";
+import Submission from "../../../models/Submission";
+import Verdict from "../../../models/Verdict";
+import './ExamSubmission.scss';
+
+const ExamSubmission = () => {
+    const { url: currentURL } = useRouteMatch();
+    const { currentExam } = useExamContext();
+    const { examId } = useParams();
+    const [submissions] = useState([
```
Comment (Johnny850807) [inline src/app/exam/submission/ExamSubmission.js:18]: ` const [submissions, setSubmissions]`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]

File: src/models/Verdict.js [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
+        this.grade = grade;
+        this.maxGrade = maxGrade;
+    }
+}
+
+export class JudgeStatus {
+    constructor({ fullName }) {
+        this.fullName = fullName;
+    }
+}
```
Comment (Johnny850807) [inline src/models/Verdict.js:29]: This class is not needed

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done.

**Review by Johnny850807 — APPROVED** [review]
LGTM

File: src/app/exam/submission/ExamSubmission.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                return <span>({verdict.maximumRuntime}&nbsp;ms,&nbsp;{verdict.maximumMemoryUsage}&nbsp;KB)</span>
+            case 'WA':
+            case 'TLE':
+            case 'RE':
+                return <span>(score:&nbsp;{verdict?.totalGrade})</span>
+            case 'CE':
+            default:
+                return <span />;
+        }
+    }
```
Comment (Johnny850807) [inline src/app/exam/submission/ExamSubmission.js:56]:
```js
function renderStatus(verdict) {
    /*switch case*/
}

const VerdictSummary = ({verdict}) => {
   ...
  
   {renderStatus(verdict)}
    ...
}
```

**Review by Chaoyuuu — APPROVED** [review]
LGTM

File: src/app/exam/submission/ExamSubmission.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                    ]}
+                    tableRowGenerator={{
+                        list: verdictIssuedEvents,
+                        key: (verdictIssuedEvent) => verdictIssuedEvent.problemId,
+                        data: (verdictIssuedEvent) => [
+                            <TableCell>{verdictIssuedEvent?.problemId}</TableCell>,
+                            <FakeLink>{verdictIssuedEvent?.problemTitle}</FakeLink>,
+                            <TableCell>{verdictIssuedEvent?.studentId}</TableCell>,
+                            <TableCell>
+                                <VerdictDescription>{verdictIssuedEvent}</VerdictDescription>
```
Comment (Johnny850807) [inline src/app/exam/submission/ExamSubmission.js:82]: `<VerdictSummary verdict={verdictIssuedEvent.verdict}/>`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: src/app/exam/submission/ExamSubmission.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            case 'TLE':
+            case 'RE':
+                return <span>(score:&nbsp;{verdict?.totalGrade})</span>
+            case 'CE':
+            default:
+                return <span />;
+        }
+    }
+
+    return <div className="exam-submissions">
```
Comment (nyngwang) [inline src/app/exam/submission/ExamSubmission.js:55]: I think this should match the class name.

**Review by nyngwang — COMMENTED** [review]
Review Done!

File: src/app/exam/submission/ExamSubmission.js (hunk 5 above)
Comment (Johnny850807) [reply src/app/exam/submission/ExamSubmission.js:55]: nice catch

**Review by nyngwang — APPROVED** [review]
LGTM.

File: src/app/exam/submission/ExamSubmission.js (hunk 3 above)
Comment (Johnny850807) [reply src/app/exam/submission/ExamSubmission.js]: @Wally5077 unresolved

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**